App/src/services/group.ts

```diff
@@ -0,0 +1,44 @@
+//This file is used to define group related types and apis
+import axios from "axios";
+import { User } from "./user";
+
+//types
+export interface GroupDTO {
+    id: string | null;
+    name: string | null;
+    members: User[] | null;
+}
+
+export interface GroupCreateDTO {
+    name: string | null;
+}
+
+export interface GroupResponseDTO {
+    data: GroupDTO;
+}
+
+export interface MultiGroupResponseDTO {
+    data: GroupDTO[];
+}
+
+//api calls
+export function CreateGroupCall(dto: GroupCreateDTO): Promise<GroupResponseDTO> {
+    return axios.post("/Groups/CreateGroup", dto);
+}
+
+export function GetGroupByID(id: string): Promise<GroupResponseDTO> {
+    return axios.get("/Groups/GetGroupById/" + id);
```

Comment:
Nice organization in the service file. These routes are all wrong though, because the base part of the url should be `Group`, not `Groups`. Also for get group by id, the route does not have `GetGroupById`, it's just `/Group/:id`

Reply:
Does the semicolon have to be there for /Group/:id or is it just a separator for where the string ends and the id is added?